js/character-counter.js

```diff
@@ -3,44 +3,68 @@
 	var $ = w.jQuery;
 
 	function CharacterCounter( el ) {
-		this.characterCounter.call( el );
+		var $el = $(el);
+
+		if( $el.data("CharacterCounter") ){
+			return;
+		}
+		$el.data( "CharacterCounter", this );
+
+		var self = this;
+
+		this.$el = $el;
+		this.$label = $(
+			"[data-maxlength-count='" + this.$el.attr("id") + "']," +
+				"[data-minlength-count='" + this.$el.attr("id") + "']"
+		);
+
+		this.$parent = this.$label.parent().addClass( "character-counter plural" );
+		this.max = parseFloat( this.$el.attr( "maxlength" ), 10 );
+		this.min = parseFloat( this.$el.attr( "minlength" ), 10 );
+
+		if ( this.$label.length ){
+			this.$el
+				.unbind( ".charcount")
+				.bind( "input.charcount keyup.charcount", function(){
+					self.characterCount();
+				});
+		}
```

Comment:
Note that I've moved a whole bunch of the `characterCount` method stuff to the constructor meaning it gets done once and only once. I can't see that being an issue but it's worth a double check.